lib/email/receiver.rb

```diff
@@ -589,11 +589,10 @@ def parse_from_field(mail = nil)
       # which we can extract from embedded_email_raw.
       if has_been_forwarded?
         if mail[:from].to_s =~ group_incoming_emails_regex && embedded_email[:from].errors.blank?
-          embedded_email[:from].each do |address_field|
-            from_address = address_field.address
-            from_display_name = address_field.display_name&.to_s
-            next if !from_address&.include?("@")
-            return [from_address&.downcase, from_display_name&.strip]
+          from_address, from_display_name = extract_from_fields_from_header(embedded_email, :from)
+          if from_address
+            @forwarded_by_address, @forwarded_by_name = extract_from_fields_from_header(mail, :from)
```

Comment:
I am not sure if @forwarded_by_address and @forwarded_by_name should be instance variables as they are quite specific to 2 methods.

You already have instance variable @mail which makes sens because whole class is using it. What would you say for some helper method like
```
def fowarded_by_address_and_forwarded_by_name
  @fowarded_by_address_and_forwarded_by_name ||= extract_from_fields_from_header(@mail, :from)
end
def fowarded_by_address
  @forwarded_by_address ||= fowarded_by_address_and_forwarded_by_name&.first
end
def fowarded_by_name
  @forwarded_by_name ||= fowarded_by_address_and_forwarded_by_name&.last
end

# This instance variables are just to memoize, not to be used
```

Reply:
Yes I think this is a good call, will change